src/Network/ABCI.hs

```diff
@@ -21,74 +22,41 @@ import           Network.ABCI.Types.App               (App (..))
 import qualified Network.ABCI.Types.Error             as Error
 import qualified Network.ABCI.Types.Messages.Request  as Request
 import qualified Network.ABCI.Types.Messages.Response as Response
-import           Network.Socket                       (SockAddr)
 import qualified Proto.Types                          as PT
 import qualified Proto.Types_Fields                   as PT
-import           UnliftIO                             (MonadUnliftIO)
-
 
 -- | Default ABCI app network settings.
 defaultSettings :: ServerSettings
 defaultSettings = serverSettings 26658 $ fromString "127.0.0.1"
 
+type AppRunner m = BS.ByteString -> m BS.ByteString
+
 -- | Serve an ABCI application with custom 'ServerSettings'
-serveAppWith
-  :: ( MonadIO m
-     , MonadBaseControl IO m
-     , MonadUnliftIO m
-     )
-  => ServerSettings
-  -> (SockAddr -> m (App m))
-  -> m ()
-serveAppWith cfg mkApp = runGeneralTCPServer cfg $ \appData -> do
-  app <- mkApp (appSockAddr appData)
-  runConduit (setupConduit app appData)
+serveAppWith :: ServerSettings -> AppRunner IO -> IO ()
+serveAppWith cfg runner = runTCPServer cfg $ \appData -> do
+  runConduit $
+      appSource appData
+    .| CL.mapM runner
+    .| appSink appData
 
 -- | Serve an ABCI application with default 'ServerSettings'
-serveApp
-  :: ( MonadIO m
-     , MonadBaseControl IO m
-     , MonadUnliftIO m
-     )
-  => (SockAddr -> m (App m))
-  -> m ()
+serveApp :: AppRunner IO -> IO ()
 serveApp = serveAppWith defaultSettings
 
 
-
-    -- | Sets up the application wire pipeline.
-setupConduit
-  :: ( MonadIO m
-     , MonadUnliftIO m
-     )
-  => App m
-  -> AppData
-  -> ConduitT i o m ()
-setupConduit app appData =
-     appSource appData
-  .| Wire.decodeLengthPrefixC
-  .| CL.map (traverse decodeRequestValue =<<)
-  .| CL.mapM (respondWith app)
-  .| CL.map (map PL.encodeMessage)
-  .| Wire.encodeLengthPrefixC
-  .| appSink appData
+runApp :: Monad m => App m -> AppRunner m
```

Comment:
With current arrangement I would say to even move runApp and decodeLengthPrefix/encodeLengthPrefix to App module